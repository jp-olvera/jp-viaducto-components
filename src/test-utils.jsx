import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
