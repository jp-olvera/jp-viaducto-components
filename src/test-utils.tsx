import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
