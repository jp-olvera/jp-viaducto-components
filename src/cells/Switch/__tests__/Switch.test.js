import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '../../../test-utils';
import { Switch } from '..';

describe('<Switch/>', () => {
  test('should render properly', () => {
    render(<Switch id="Switch" />);
    expect(screen.getByTestId('Switch')).toBeVisible();
  });

  test('should be selected', () => {
    const { getByTestId } = render(<Switch id="Switch" />);
    const input = getByTestId('Switch');
    const selector = input.querySelector('input');
    fireEvent.click(selector);
    expect(selector.checked).toEqual(true);
  });

  test('should be not checked', () => {
    const { getByTestId } = render(<Switch id="switch1" />);
    const input = getByTestId('switch1');
    const selector = input.querySelector('input');
    expect(selector.checked).toEqual(false);
    fireEvent.click(selector);
    expect(selector.checked).toEqual(true);
  });

  test('should be disabled', () => {
    const { getByTestId } = render(<Switch id="Switch1" disabled />);
    expect(
      getByTestId('Switch1').querySelector('input').hasAttribute('disabled'),
    ).toEqual(true);
  });
});
