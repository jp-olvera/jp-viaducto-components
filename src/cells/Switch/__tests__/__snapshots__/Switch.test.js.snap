// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch/> should fire hover event 1`] = `
<div>
  <label
    class="StyledSwitch-sc-10hvevw-0 hDDaiJ"
    color="#9060EB"
  >
    <input
      type="checkbox"
    />
    <span
      class="slider round"
      data-testid="slider"
    />
  </label>
</div>
`;
